src/osm-gps-map/osm-gps-map-qt.cpp

```diff
@@ -219,6 +221,14 @@ Maep::GpsMap::GpsMap(QQuickItem *parent)
   bool orientation = gconf_get_bool(GCONF_KEY_SCREEN_ROTATE, TRUE);
   int gpsRefresh = gconf_get_int(GCONF_KEY_GPS_REFRESH_RATE, 1000);
   bool compassEnabled = gconf_get_bool(GCONF_KEY_COMPASS_ENABLED, FALSE);
+  gint compassMode = std::min(static_cast<gint>(COMPASS_N_MODES) - 1,
+                              std::max(static_cast<gint>(COMPASS_MODE_OFF),
+                                        gconf_get_int(GCONF_KEY_COMPASS_MODE,
+                                                      static_cast<gint>(COMPASS_MODE_OFF))));
+
+  // backward compatibility with old boolean setting
+  if (compassEnabled && compassMode == COMPASS_MODE_OFF)
+    compassMode = COMPASS_MODE_NORTH;
```

Comment:
Made a new commit that does both, i.e. use the nicer way of reading the config *and* unset the deprecated setting in the d'tor.